### ISSUE TITLE
Memory Consumption in IoS

### ISSUE THREAD

**Issue by aftabaminzoobiapps** [ammarahm-ed/react-native-admob-native-ads#107]
I am using this library. and when I show ads on different screens then there is an issue. with navigation to a screen where there I am showing native ad the memory is increasing by 2MB. Here is my Code. 
           `import React, {useState, useEffect, useRef} from 'react';
import { Platform, View, Image, Dimensions} from 'react-native';
import NativeAdView, {CallToActionView,HeadlineView,IconView,StarRatingView, TaglineView,StoreView, AdBadge,} from 'react-native-admob-native-ads';
    let NATIVE_AD_ID = "xxxxxx-yyyyy-zzzzzzzz";
    const delay = 0
    const WIDTH = Dimensions.get('screen').width ;
        export const AdView = (props) => {
            return (
                <NativeAdView
                    onAdLoaded={() => {}}
                    onAdFailedToLoad={() => {}}
                    onAdLeftApplication={() => { }}
                    onUnifiedNativeAdLoaded={(data) => {}}
                    delayAdLoading={delay}
                    refreshInterval={700000 * 2}
                    style={{
                      width: WIDTH,
                      alignSelf: 'center',
                      marginVertical: 0,
                      backgroundColor:"#fff",
                    }}
                    adUnitID={ NATIVE_AD_ID} // REPLACE WITH NATIVE_AD_VIDEO_ID for video ads.
                  >
                  <View
                    style={{width: '100%'}}>
                    <View
                      style={{
                        height: 70,
                        width: '100%',
                        flexDirection: 'row',
                        justifyContent: 'space-between',
                        alignItems: 'center',
                        paddingHorizontal:0,
                        backgroundColor:"transparent"
                      }}>
                      <View  style={{
                          width: 70,
                          height: 70,
                        }}>
                      <AdBadge 
                        textStyle={{color:'#fff'}}
                          style={{marginLeft:1,zIndex:3, marginTop:1, backgroundColor:"orange", borderColor:'#fff', 
                          width:20 , alignItems:"center", position:'absolute'}}
                      />
                      
                      <IconView
                        style={{
                          width: 70,
                          height: 70,
                        }}
                      />
                      </View>
                      <View
                        style={{
                          width: '60%',
                          maxWidth: '60%',
                          backgroundColor:"#fff",
                          marginLeft:3,
                          paddingHorizontal: 2,
                        }}>
                        <HeadlineView
                          hello="abc"
                          style={{
                            marginTop:2,
                            fontWeight: 'bold',
                            fontSize: 8,
                          }}
                        />
                        <TaglineView
                          numberOfLines={2}
                          style={{
                            fontSize: 8,
                            backgroundColor:"#fff"
                          }}
                        />
                      
                        <View style={{width:'60%', height:2,}}></View>

                        <View
                          style={{
                            flexDirection: 'row',
                            alignItems: 'center',
                            backgroundColor:"#fff"
                          }}>
                          <StarRatingView
                            starSize={12}
                            fullStarColor="orange"
                            emptyStarColor="gray"
                            containerStyle={{
                              width: 65,
                              marginTop: 4,
                            }}
                          />

                          <StoreView
                            style={{
                              fontSize: 12,
                              marginLeft: 10,
                              backgroundColor:'#fff'
                            }}
                          />
                        </View>
                      </View>
                      <CallToActionView
                        style={{
                          height: 25,
                          backgroundColor:"red",
                          paddingHorizontal: 12,
                          backgroundColor: 'purple',
                          justifyContent: 'center',
                          alignSelf:'center',
                          alignItems: 'center',
                          borderRadius: 5,
                          elevation: 1,
                          marginRight:2,
                        }}
                        textStyle={{color: 'white', fontSize: 8}}
                      />
                    </View>
                  </View>
                </NativeAdView>
              );
        };
`

HERe how I am using this in my screens 
`import React from 'react';
import {SafeAreaView,View,Text,StatusBar, Pressable } from 'react-native';
import  {AdView}  from '../components/NativeAdd'
import styles ,{appcolor} from "../styles/styles";
const App = ({navigation}) => {
  return (
    <View style={{flex:1, justifyContent:"center", alignItems:"center"}}> 
      <StatusBar barStyle="dark-content" />
      <SafeAreaView>
        <View style={styles.footer}>
          <AdView />
        </View>
      </SafeAreaView>
    </View>
  )
}
export default App;`

After some time the app crashes out to memory usage.

**Comment by aftabaminzoobiapps** [comment]
for details please check the example file and memory usage. I am using iPhone 6 .

**Comment by ammarahm-ed** [comment]
Please explain your issue in detail, does the app crash when showing alot of ads or is it that the ads are not cleared from memory, so you keep navigating and the memory keeps on increasing by 2 MB. Provide a simple example so I can debug this.

**Comment by aftabaminzoobiapps** [comment]
> Please explain your issue in detail, does the app crash when showing a lot of ads, or is it that the ads are not cleared from memory, so you keep navigating and the memory keeps on increasing by 2 MB. Provide a simple example so I can debug this.

as I explain in the issue. running the example in the Github repository has also this issue. ads are not cleaning from memory.

**Comment by aftabaminzoobiapps** [comment]
Any updates for this.

**Comment by Terfender** [comment]
When adding only 3 native ads on main screen using the example blow, the app becomes really really slow. When removing them, the app works fine.


```
        <NativeAdView
          adUnitID={unitId}>
          <MediaView style={styles.banner} />
          <View style={styles.view}>
            <HeadlineView style={styles.title} />
          </View>
          ```

**Comment by aftabaminzoobiapps** [comment]
> When adding only 3 native ads on main screen using the example blow, the app becomes really really slow. When removing them, the app works fine.
> 
> ```
>         <NativeAdView
>           adUnitID={unitId}>
>           <MediaView style={styles.banner} />
>           <View style={styles.view}>
>             <HeadlineView style={styles.title} />
>           </View>
>           ```
> ```

this is because the native adds or consuming a lot of memory. and is not releasing.i tried my best to resolve this issue but didn't found any one.

**Comment by ammarahm-ed** [comment]
@aftabaminzoobiapps I will look into it. I have investigated it a bit but couldn't find the bug. needs more investigation. Can you provide an example of exactly how to reproduce the issue and how many ads did you load?

**Comment by aftabaminzoobiapps** [comment]
> @aftabaminzoobiapps I will look into it. I have investigated it a bit but couldn't find the bug. needs more investigation. Can you provide an example of exactly how to reproduce the issue and how many ads did you load?

I have ads on multiple screens. I have a test that on iPhone 6 real device and iPhone 9 and iPhone X simulators. if I don't load any ads the memory is the consumption is the same. but when I add native Ad from this library the memory consumption is increasing each time ai navigate to the screen. I have also tested [https://github.com/ammarahm-ed/react-native-admob-native-ads/tree/master/example](URL) that you have provided in the repository it has a similar problem of memory consumption. I really need to resolve this issue. because one of project is delayed due to this problem. can you please solve this please as soon as possible?

**Comment by ammarahm-ed** [comment]
@aftabaminzoobiapps Does the memory stay allocated even after destroying the ads?

**Comment by aftabaminzoobiapps** [comment]
yes sir. after continuous loading ads for 5 to 6 minutes, the app getting
slow and crashed on iPhone 6 real device.

**Comment by ammarahm-ed** [comment]
@aftabaminzoobiapps hey, can you test the master branch on iOS to check if this is fixed. I have tried to fix it.

**Comment by aftabaminzoobiapps** [comment]
ok, I will test it later.

**Comment by zoobibackups** [comment]
I have tested my app and there is still some issue. before that fix @ammarahm-ed when I was navigating through the screens per ad memory consumption was 2MB.  no, when I navigate 10 times the memory consumption was like this. before it was 193 MB and after 10 times of navigation it was about 207 MBs . and after some time it was reduced to 197 MB. so for 10 navigation  it is consuming 4 MBs.

**Comment by ammarahm-ed** [comment]
The garbage collector on iOS runs after a while I think. Maybe if we remove
the ad refreshing logic it will lower down even more. Currently this seems
to have reduced the memory consumption overall. True?

**Comment by zoobibackups** [comment]
yes. i is reduced but still to have check on real devices.

**Comment by ammarahm-ed** [comment]
fixed properly for 0.4.0 on master
